### PR TITLE
Sc47/connecting wallet

### PR DESCRIPTION
**Pull request by Justin1002**
Added login and logout functionality with Metamask integration

Logout is sort of a fake log out using local storage, but for the user experience it works fine!
Peek 2022-01-26 15-55

### CODE REVIEW

**Review by akshay-rakheja — DISMISSED** [review]

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -1,18 +1,81 @@
...
   Routes,
 } from "react-router-dom";
+import { ethers } from 'ethers';
 
 import Nav from './components/Nav'
 import Home from './components/Home'
 
 function App() {
+  const [account, setAccount] = useState(null);
+  let loggedIn = localStorage.getItem("loggedIn") || null;
```
Comment (connie-ho) [inline frontend/src/App.jsx:14]: localStorage strikes again

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -1,18 +1,81 @@
...
+  
+  const checkWalletConnection = async () => {
```
Comment (connie-ho) [inline frontend/src/App.jsx:16]: also i feel like these functions should be put in a custom hook to keep dis component clean ja feel

Comment (Justin1002) [reply frontend/src/App.jsx]: thx bb u rite

File: frontend/src/App.jsx (hunk 1 above)
Comment (Justin1002) [reply frontend/src/App.jsx]: xo

**Comment by connie-ho** [comment]
don't forget to rerequest!!!!!!!!!!!!!!!!!!!!

**Review by connie-ho — APPROVED** [review]

File: frontend/src/components/common/CardItem.jsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ const CardItem = ({ children, onClick, rootClassName, ...props }) => {
 
   return (
     <Card 
-      boxShadow={4}
+      boxshadow={4}
```
[the remark this answers is not in the document]
Comment (Justin1002) [reply frontend/src/components/common/CardItem.jsx:31]: no but it was giving me an error :3

File: frontend/src/__tests__/App.test.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { render} from '@testing-library/react';
+import ReactDOM from 'react-dom';
```
[the remark this answers is not in the document]
Comment (Justin1002) [reply frontend/src/__tests__/App.test.jsx:2]: test driven development bb

File: frontend/src/components/common/NavButton.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+import { Button } from '@mui/material';
+
+const NavButton = ({ children, onClick, rootClassName, ...props }) => {
+
+  return (
+    <Button 
+      variant="text"
+      style={{
```
Comment (Sufiyan1997) [inline frontend/src/components/common/NavButton.jsx:10]: Not sure about this. But shouldn't we add this CSS via mui?

Comment (Justin1002) [reply frontend/src/components/common/NavButton.jsx:10]: you're right but for some reason it wasn't overwriting the styles for the button, I'll need to look into further but anyways this is just an interim button for now which will likely be deleted so I didn't want to spend too much time on it

**Pull request merged**